test/addons-napi/test_promise/test_promise.c

```diff
@@ -0,0 +1,55 @@
+#include <node_api.h>
+#include "../common.h"
+
+napi_ref deferred_ref = NULL;
+
+napi_value createPromise(napi_env env, napi_callback_info info) {
+  napi_value promise, deferred;
+
+  if (deferred_ref != NULL) {
+    NAPI_CALL(env, napi_delete_reference(env, deferred_ref));
+  }
+
+  NAPI_CALL(env, napi_create_promise(env, &deferred, &promise));
+  NAPI_CALL(env, napi_create_reference(env, deferred, 1, &deferred_ref));
+
+  return promise;
+}
+
+napi_value concludeCurrentPromise(napi_env env, napi_callback_info info) {
+  napi_value argv[2], deferred;
+  size_t argc = 2;
+  bool resolution;
+
+  NAPI_CALL(env, napi_get_cb_info(env, info, &argc, argv, NULL, NULL));
+  NAPI_CALL(env, napi_get_value_bool(env, argv[1], &resolution));
+  NAPI_CALL(env, napi_get_reference_value(env, deferred_ref, &deferred));
+  NAPI_CALL(env, napi_conclude_deferred(env, deferred, argv[0], resolution));
+
+  return NULL;
+}
+
+napi_value isPromise(napi_env env, napi_callback_info info) {
+  napi_value promise, result;
+  size_t argc = 1;
+  bool is_promise;
+
+  NAPI_CALL(env, napi_get_cb_info(env, info, &argc, &promise, NULL, NULL));
+  NAPI_CALL(env, napi_is_promise(env, promise, &is_promise));
+  NAPI_CALL(env, napi_get_boolean(env, is_promise, &result));
+
+  return result;
+}
+
+void Init(napi_env env, napi_value exports, napi_value module, void* priv) {
+  napi_property_descriptor descriptors[] = {
+    DECLARE_NAPI_PROPERTY("createPromise", createPromise),
+    DECLARE_NAPI_PROPERTY("concludeCurrentPromise", concludeCurrentPromise),
+    DECLARE_NAPI_PROPERTY("isPromise", isPromise),
```

Comment:
There should be an anti-case for passing a promise to `napi_conclude_deferred()`.